mods/ra/maps/production-disruption/production-disruption.lua

```diff
@@ -0,0 +1,270 @@
+		Actor.Create("ctnk", true, { Owner = greece, Location = ChronoSpawn3.Location })
+		Actor.Create("camera", true, { Owner = greece, Location = EasyCamera.Location })
+		Media.PlaySound("chrono2.aud")
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local sovietWaterSquad2 = Reinforcements.ReinforceWithTransport(ussr, "lst", SovietSquad, { WaterEntry2.Location, Beach3.Location }, { WaterEntry2.Location })[2]
+		Utils.Do(sovietWaterSquad2, function(a)
+			Trigger.OnAddedToWorld(a, function()
+				a.AttackMove(FlameCam.Location)
+				IdleHunt(a)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(13), function()
+		local sovietWaterSquad2 = Reinforcements.ReinforceWithTransport(ussr, "lst", V2Squad, { WaterEntry2.Location, Beach2.Location }, { WaterEntry2.Location })[2]
+		Utils.Do(sovietWaterSquad2, function(a)
+			Trigger.OnAddedToWorld(a, function()
+				a.AttackMove(FlameCam.Location)
+				IdleHunt(a)
+			end)
+		end)
+	end)
+end
+
+MissileSubEscape = function()
+	missileSub = Actor.Create("msub", true, { Owner = ussr, Location = MissileSubSpawn.Location })
+	subEscapeCam = Actor.Create("camera", true, { Owner = greece, Location = SubPath2.Location })
+	DestroySub = greece.AddPrimaryObjective("Destroy the submarine before it escapes!.")
+
+	Utils.Do(SubEscapePath, function(waypoint)
+		missileSub.Move(waypoint.Location)
+	end)
+
+	Trigger.OnEnteredFootprint({ SubPath3.Location }, function(a, id)
+		if a.Owner == ussr and a.Type == "msub" then
+			Trigger.RemoveFootprintTrigger(id)
+			ussr.MarkCompletedObjective(ussrObj)
+		end
+	end)
+
+	Trigger.OnKilled(missileSub, function()
+		greece.MarkCompletedObjective(DestroySub)
+	end)
+end
+
+FinishTimer = function()
+	for i = 0, 5 do
+		local c = TimerColor
+		if i % 2 == 0 then
+			c = HSLColor.White
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(i), function() UserInterface.SetMissionText("The sub is heading for open sea!", c) end)
+	end
+	Trigger.AfterDelay(DateTime.Seconds(6), function() UserInterface.SetMissionText("") end)
+end
+
+UnitsArrived = false
+TimerFinished = false
+ticked = TimerTicks
+Tick = function()
+	if badguy.PowerState ~= "Normal" then
+		PowerDownTeslas()
+	end
+
+	if greece.HasNoRequiredUnits() and UnitsArrived == true then
+		ussr.MarkCompletedObjective(ussrObj)
+	end
+
+	if ticked > 0 then
+		UserInterface.SetMissionText("Submarine competes in " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+	elseif ticked == 0 and not TimerFinished == true then
+		MissileSubEscape()
+		TimerFinished = true
+	end
+end
+
+WorldLoaded = function()
+	greece = Player.GetPlayer("Greece")
+	ussr = Player.GetPlayer("USSR")
+	badguy = Player.GetPlayer("BadGuy")
+
+	ussrObj = ussr.AddPrimaryObjective("Get a missile sub to open waters.")
+	StopProduction = greece.AddPrimaryObjective("Destroy the Soviet sub pen.")
+	PowerDownTeslaCoils = greece.AddSecondaryObjective("Take down power to the tesla coils.")
```

Comment:
This should likely be a primary objective, considering that you can't progress without doing it.